code-challenges.test.js

```diff
@@ -18,25 +23,64 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
+// describe the name of the function and use variables provided.
+describe("shuffleArray", () => {
+    let colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    let outPut1 = ["yellow", "blue", "pink", "green"]
+    let colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    let outPut2 = ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+    // describe what the function does.
+    test("returns first item from the array and shuffles the remaining content.", () => {
+  
+      //check the expected outputs
+      expect(shuffleArray(colors1)).toEqual(outPut1)
+      expect(shuffleArray(colors2)).toEqual(outPut2)
+    })
+  })
+
 var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
-
-
-
 // b) Create the function that makes the test pass.
 
-
+//declare a function named shuffleArray
+//place a parameter names array
+const shuffleArray = (array) => {
+    //at this point I had trouble remembering how to use higher order functions so I had to go back and review the syllabus.
+    //create a variable that maps through the array
+       let eachString = array.map(value => {
```

Comment:
remember the map is accessing each item in the array one at a time. SO if you try to call on [0] inside of the map it's gonna break. Break this problem down into smaller bits and pseudo code it out. I want to see your full logic and then see your code.